### ISSUE TITLE
findOneAndUpdate option "new" has different default value from MongoDB findAndModify

### ISSUE THREAD

**Issue by mocheng** [LearnBoost/mongoose#2262]
In Mongoose doc http://mongoosejs.com/docs/api.html#model_Model.findOneAndUpdate
> new: bool - true to return the modified document rather than the original. defaults to true

In MongoDB doc http://docs.mongodb.org/manual/reference/method/db.collection.findAndModify/
>new	Boolean	Optional. When true, returns the modified document rather than the original. The findAndModify() method ignores the new option for remove operations. The default is false.

The mongoose default value of `options.new` is true while MongoDB is `false`.
This is really counter-intuitive.

Any reason for this API design?

**Comment by Shuliyey** [comment]
can you return both?

**Comment by Shuliyey** [comment]
can you return both original and new document for comparison purpose?

**Comment by vkarpov15** [comment]
@Shuliyey there's no way to do so that I'm aware of. Would be a useful feature though :)

**Comment by mnort9** [comment]
This change makes the api less expressive IMO and will likely trip up newcomers. I believe most people will expect 'findOneAndUpdate' to return the updated doc. I can understand the desire to mirror the MongoDB driver, but creating an expressive high level api should take precedence.

Our dev team recently upgraded mongoose and we've had to go back to add what seems to be a boilerplate condition ({new: true}) to all of our 'findOneAndUpdate' queries. This seems like a step backward.

I could be missing something, but what scenario would you not want the updated doc to be returned?

**Comment by vkarpov15** [comment]
@mnort9 one common use case for `findOneAndUpdate` is as a distributed locking / user-implemented transaction mechanism. For instance, suppose you want to do something like "switch job's state to 'running', but I also want to know if this job was in the 'pending' state before it was changed" in order to ensure that only one process runs a given job. In that case, you want `new: false`. Admittedly we don't have good information on whether `new: true` or `new: false` is used more frequently, so that's why we changed the default for 4.0: we don't have a solid basis for diverging from the mongodb API in such a big way.
